plugins/libraries.ts

```diff
@@ -0,0 +1,66 @@
+import { Plugin } from '@nuxt/types/app';
+
+type LibrariesType = {
+  fetchItem: (id: string) => Promise<void>;
+  fetchItems: (params: Record<string, unknown>) => Promise<string[]>;
+};
+
+declare module '@nuxt/types' {
+  interface Context {
+    $libraries: LibrariesType;
+  }
+
+  interface NuxtAppOptions {
+    $libraries: LibrariesType;
+  }
+}
+
+declare module 'vue/types/vue' {
+  interface Vue {
+    $libraries: LibrariesType;
+  }
+}
+
+declare module 'vuex/types/index' {
+  // eslint-disable-next-line -- Current TypeScript rules flag S as unused, but Nuxt requires identical types
+  interface Store<S> {
+    $libraries: LibrariesType;
+  }
+}
+
+const librariesPlugin: Plugin = ({ $api, $auth, store }, inject) => {
+  inject('libraries', {
+    /**
+     * Fetches an item based on its ID and stores it
+     *
+     * @param {string} id - Item ID
+     */
+    fetchItem: async (id: string): Promise<void> => {
+      const item = (
+        await $api.userLibrary.getItem({
+          userId: $auth.user?.Id,
+          itemId: id
+        })
+      ).data;
+      await store.dispatch('items/addItem', { item });
+    },
+
+    /**
+     * Executes a getItems API call with given parameters, stores the result and returns the ID list
+     *
+     * @param {object} params - Parameters of getItems API call (without user ID)
+     * @returns {string[]} list of IDs
+     */
+    fetchItems: async (params: Record<string, unknown>): Promise<string[]> => {
+      const result = (
+        await $api.items.getItems({ ...params, userId: $auth.user?.Id })
+      ).data.Items;
+      if (!result) return [];
+      await store.dispatch('items/addItems', { items: result });
+      // @ts-expect-error - stupid parser thinking it'll be an (undefined|string)[] when I explicitly filter them
```

Comment:
This could use a rewording to avoid the first person and the "informal" language :yum: 

Something like "TypeScript badly infers types here, we are filtering out undefined Ids" or something.

Reply:
:(